internal/provider/cluster_resource.go

```diff
@@ -1605,3 +1733,41 @@ func reconcileRegionUpdate(
 func NewClusterResource() resource.Resource {
 	return &clusterResource{}
 }
+
+func byocEqual(a *CustomerCloudAccount, b *CustomerCloudAccount) bool {
+	if a == nil && b == nil {
+		return true
+	}
+	if (a == nil) != (b == nil) {
+		return false
+	}
+	// both non-nil
+	if (a.Aws == nil) != (b.Aws == nil) {
+		return false
+	}
+	if a.Aws != nil && b.Aws != nil {
+		if a.Aws.Arn.ValueString() != b.Aws.Arn.ValueString() {
```

Comment:
nit:  It would probably be a bit more readable to have separate test functions for each of aws, gcp and azure and then compose them here.